Packs/PassiveTotal/Integrations/PassiveTotal_v2/PassiveTotal_v2.py

```diff
@@ -0,0 +1,871 @@
+    :return: standard command result
+    """
+    # Retrieve arguments
+    params = get_common_arguments(args)
+    params['timeout'] = args.get('timeout', 7)
+
+    # http call
+    resp = client.http_request("GET", url_suffix=URL_SUFFIX['GET_PDNS_DETAILS'], params=params)
+    total_record = resp.get('totalRecords', 0)
+    if total_record <= 0:
+        return MESSAGES['NO_RECORDS_FOUND'].format('PDNS Record(s)')
+
+    # Creating entry context
+    results = resp.get('results', [])
+    custom_ec = createContext(results, removeNull=True)
+    standard_ec = create_pdns_standard_context(results)
+
+    # Creating human-readable
+    hr = get_pdns_details_hr(results, total_record)
+
+    # Building custom output path
+    output_path = 'PassiveTotal.PDNS(val.{0} == obj.{0} && val.{1} == obj.{1} && val.{2} == obj.{2})' \
+        .format('resolve', 'recordType', 'resolveType')
+
+    return CommandResults(
+        outputs_prefix=output_path,
+        outputs_key_field='',
+        indicators=standard_ec,
+        outputs=custom_ec,
+        readable_output=hr,
+        raw_response=resp
+    )
+
+
+@logger
+def get_host_pairs_command(client: Client, args: Dict[str, Any]) -> Union[str, CommandResults]:
+    """
+    Retrieves the host attribute pairs related to a domain or IP address.
+
+    :param client: Client object
+    :param args: The command arguments provided by user
+    :return: Standard command result or no records found message
+    """
+    # Trim the arguments
+    for argument in args:
+        args[argument] = args[argument].strip()
+
+    # Retrieve arguments and prepare query data
+    params = get_common_arguments(args)
+
+    if args.get('direction') not in VALID_DIRECTION_FOR_HOST_PAIRS:
+        raise ValueError(MESSAGES['INVALID_DIRECTION_VALUE'])
+    params['direction'] = args.get('direction')
+
+    # http call
+    resp = client.http_request(method="GET", url_suffix=URL_SUFFIX['GET_HOST_PAIRS'], params=params)
+
+    total_records = resp.get('totalRecords', 0)
+    if total_records == 0:
+        return MESSAGES['NO_RECORDS_FOUND'].format('host pair(s)')
+
+    results = resp.get('results', [])
+
+    # Creating entry context
+    custom_ec = createContext(data=results, removeNull=True)
+
+    # Creating human-readable
+    hr = get_host_pairs_hr(results)
+
+    return CommandResults(
+        outputs_prefix='PassiveTotal.HostPair',
+        outputs_key_field='',
+        outputs=custom_ec,
+        readable_output=hr,
+        raw_response=resp
+    )
+
+
+def main() -> None:
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    # Commands dict
+    commands = {
+        'pt-ssl-cert-search': ssl_cert_search_command,
+        'pt-get-pdns-details': get_pdns_details_command,
+        'pt-whois-search': pt_whois_search_command,
+        'pt-get-components': get_components_command,
+        'pt-get-trackers': get_trackers_command,
+        'pt-get-host-pairs': get_host_pairs_command
```

Comment:
in the original PassiveTotal integration there are enrichment commands `ip, domain, url`, it will not exist in the new version?
And I see that you return DBotScore object. DBotScore has special meaning when you return it as part of the enrichment commands. Otherwise they are just objects in the context

Reply:
yes @anara123 we haven't added enrichment commands in new version v2. We did inform customer about this and they're ok with it.

We're returning DBotScore object because [CommonServerPython](https://github.com/demisto/content/blob/751eec62d98829423eb87b9e6920a873a57a67d5/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L2385) force us to create DBotScore object whenever we add anything in StandardContext

Reply:
> @crestdatasystems As discussed, a reputation command for domain will be added to the integration.
> After that, we are good to go

added reputation command